### PR TITLE
Add analytics calls to AdyenContext

### PR DESCRIPTION
**Pull request by descorp**
# Open PR

## Changes

* `AdyenContext` now sends a telemetry event when the shopper starts a payment.

### CODE REVIEW

**Review by goergisn — DISMISSED** [review]

File: src/Core/Analytics/post-telemetry.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import { NativeModules, Platform } from 'react-native';
+import { httpPost } from '../Services/https';
+import { AnalyticsProps } from './Analytics';
+let project = require('./../../../package.json');
```
Comment (ribeiroguilherme) [inline src/Core/Analytics/post-telemetry.ts]: `const`

File: src/Core/Analytics/Analytics.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+ */
+function confirmSessionDurationIsMaxFifteenMinutes(
+  checkoutAttemptIdSession: CheckoutAttemptIdSession
+): boolean {
+  if (!checkoutAttemptIdSession?.id) return false;
+
+  const fifteenMinInMs = 1000 * 60 * 15;
+  const fifteenMinAgoTimestamp = Date.now() - fifteenMinInMs;
+  return checkoutAttemptIdSession.timestamp > fifteenMinAgoTimestamp;
+}
```
Comment (ribeiroguilherme) [inline src/Core/Analytics/Analytics.ts]: Not sure that you need this? Did you take a look into the COWEB ticket? It might be relevant only for web

**Review by ribeiroguilherme — DISMISSED** [review]

File: src/Core/Analytics/post-telemetry.ts (hunk 1 above)
Comment (descorp) [reply src/Core/Analytics/post-telemetry.ts]: Fixed [776e262](https://github.com/Adyen/adyen-react-native/pull/230/commits/776e26244224590e86483fdda2680a8f572b1d04)

File: src/Core/Analytics/Analytics.ts (hunk 2 above)
Comment (descorp) [reply src/Core/Analytics/Analytics.ts]: Fixed [776e262](https://github.com/Adyen/adyen-react-native/pull/230/commits/776e26244224590e86483fdda2680a8f572b1d04)

File: src/Core/Analytics/post-telemetry.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    errorLevel: 'silent' as const,
+    environment: config.environment,
+    path: `v2/analytics/log?clientKey=${config.clientKey}`,
+  };
+
+  const constants = Platform.constants as any;
+  const telemetryEvent = {
+    amountValue: config.amount?.value,
+    amountCurrency: config.amount?.currency,
+    version: project.version,
```
Comment (ribeiroguilherme) [inline src/Core/Analytics/post-telemetry.ts]: Should be in another format.  Check [here](https://github.com/Adyen/adyen-web/pull/2219/files#diff-6594e49946aa8de03aa2a4f722d454f5c6ef7fcff93869f2e555d0c868ae166bR26)

**Review by ribeiroguilherme — CHANGES_REQUESTED** [review]

File: src/Core/Analytics/post-telemetry.ts (hunk 3 above)
Comment (descorp) [reply src/Core/Analytics/post-telemetry.ts]: Fixed [41e150d](https://github.com/Adyen/adyen-react-native/pull/230/commits/41e150d4239834c6fb8046839fc49a7831066c09)

**Review by ribeiroguilherme — APPROVED** [review]

**Pull request merged**